ports/stm32/dac.c

```diff
@@ -322,9 +330,17 @@ STATIC mp_obj_t pyb_dac_make_new(const mp_obj_type_t *type, size_t n_args, size_
         mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("DAC(%d) doesn't exist"), dac_id);
     }
 
-    pyb_dac_obj_t *dac = &pyb_dac_obj[dac_id - 1];
-    dac->base.type = &pyb_dac_type;
-    dac->dac_channel = dac_channel;
+    pyb_dac_obj_t *dac;
+    if (MP_STATE_PORT(pyb_dac_obj_all)[dac_id - 1] == NULL) {
+        // create new DAC object
+        dac = m_new0(pyb_dac_obj_t, 1);
```

Comment:
The DAC object does not contain any root pointers so it doesn't need to go in `MP_STATE_PORT`.  And it can stay statically allocated.  So I don't think any of this code here needs to change.

Reply:
I put the array in the root pointers, to check if DAC instance is initialized. This saves 16 bytes data that were permanently reserved, but I'll revert it.